Modules/Microsoft365DSC/DSCResources/MSFT_AADEntitlementManagementAccessPackageAssignmentPolicy/MSFT_AADEntitlementManagementAccessPackageAssignmentPolicy.psm1

```diff
@@ -0,0 +1,1684 @@
+                }
+            }
+            if ($null -ne $question.AdditionalProperties.isSingleLineQuestion)
+            {
+                $question.add("IsSingleLineQuestion",$question.AdditionalProperties.isSingleLineQuestion)
+            }
+            if ($null -ne $question.AdditionalProperties.choices)
+            {
+                $question.add("Choices",[Array]$question.AdditionalProperties.choices)
+            }
+            if ($null -ne $question.AdditionalProperties.allowsMultipleSelection)
+            {
+                $question.add("AllowsMultipleSelection",$question.AdditionalProperties.allowsMultipleSelection)
+            }
+            $question.remove('additionalProperties')
+        }
+        #endregion
+
+        #region Format CustomExtensionHandlers
+        $formattedCustomExtensionHandlers=@()
+        foreach ($customExtensionHandler in $getValue.CustomExtensionHandlers)
+        {
+            $customExt=@{
+                Id = $customExtensionHandler.Id
+                Stage = $customExtensionHandler.Stage
+                CustomExtension= @{
+                    Id = $customExtensionHandler.CustomExtension.Id
+                }
+            }
+            $formattedCustomExtensionHandlers += $customExt
+        }
+        #endregion
+
+        #region Format ExpirationDateTime
+        $formattedExpirationDateTime=$getValue.ExpirationDateTime
+        if($null -ne $getValue.ExpirationDateTime)
+        {
+            #$formattedExpirationDateTime=$getValue.ExpirationDateTime.ToString()
+        }
+        #endregion
+
+        $results = @{
+
+            #region resource generator code
+            Id                      = $getValue.Id
+            AccessPackageId         = $getValue.AccessPackageId
+            AccessReviewSettings    = $formattedAccessReviewSettings
+            CanExtend               = $getValue.CanExtend
+            CustomExtensionHandlers = $formattedCustomExtensionHandlers
+            Description             = $getValue.Description
+            DisplayName             = $getValue.DisplayName
+            DurationInDays          = $getValue.DurationInDays
+            ExpirationDateTime      = $formattedExpirationDateTime
+            Questions               = $formattedQuestions
+            RequestApprovalSettings = $formattedRequestApprovalSettings
+            RequestorSettings       = $formattedRequestorSettings
+            Ensure                  = 'Present'
+            Credential              = $Credential
+            ApplicationId           = $ApplicationId
+            TenantId                = $TenantId
+            ApplicationSecret       = $ApplicationSecret
+            CertificateThumbprint   = $CertificateThumbprint
+            ManagedIdentity         = $ManagedIdentity.IsPresent
+        }
+
+        return [System.Collections.Hashtable] $results
+    }
+    catch
+    {
+        try
+        {
+            Write-Verbose -Message $_
```

Comment:
Please update this logging logic with the same code as is currently being used in other resources. This has recently been updated to create better logging.